src/main/java/cn/edu/thu/tsfile/timeseries/write/page/PageWriterImpl.java

```diff
@@ -21,97 +27,103 @@
  *
  */
 public class PageWriterImpl implements IPageWriter {
-    private static Logger LOG = LoggerFactory.getLogger(PageWriterImpl.class);
+	private static Logger LOG = LoggerFactory.getLogger(PageWriterImpl.class);
+
+	private ListByteArrayOutputStream buf;
+	private List<ByteArrayInputStream> pageList;
+	private final Compressor compressor;
+	private final MeasurementDescriptor desc;
+
+	private long totalValueCount;
+	private long maxTimestamp;
+	private long minTimestamp = -1;
 
-    private ListByteArrayOutputStream buf;
-    private final Compressor compressor;
-    private final MeasurementDescriptor desc;
+	public PageWriterImpl(MeasurementDescriptor desc) {
+		this.desc = desc;
+		this.compressor = desc.getCompressor();
+		this.buf = new ListByteArrayOutputStream();
+		this.pageList = new ArrayList<>();
+	}
 
-    private long totalValueCount;
-    private long maxTimestamp;
-    private long minTimestamp = -1;
+	@Override
+	public void writePage(ListByteArrayOutputStream listByteArray, int valueCount, Statistics<?> statistics,
+			long maxTimestamp, long minTimestamp) throws PageException {
+		// compress the input data
+		if (this.minTimestamp == -1)
+			this.minTimestamp = minTimestamp;
+		this.maxTimestamp = maxTimestamp;
+		int uncompressedSize = listByteArray.size();
+		ListByteArrayOutputStream compressedBytes = compressor.compress(listByteArray);
+		int compressedSize = compressedBytes.size();
+		PublicBAOS tempOutputStream = new PublicBAOS(estimateMaxPageHeaderSize() + compressedSize);
+		// write the page header to IOWriter
+		try {
+			ReadWriteThriftFormatUtils.writeDataPageHeader(uncompressedSize, compressedSize, valueCount, statistics,
+					valueCount, desc.getEncodingType(), tempOutputStream, maxTimestamp, minTimestamp);
+		} catch (IOException e) {
+			resetTimeStamp();
+			throw new PageException(
+					"meet IO Exception in writeDataPageHeader,ignore this page,error message:" + e.getMessage());
+		}
+		this.totalValueCount += valueCount;
+		try {
+			compressedBytes.writeAllTo(tempOutputStream);
+		} catch (IOException e) {
+			/*
+			 * In our method, this line is to flush listByteArray to buf, both
+			 * of them are in class of ListByteArrayOutputStream which contain
+			 * several ByteArrayOutputStream. In general, they won't throw
+			 * exception. The IOException is just for interface requirement of
+			 * OutputStream.
+			 */
+			throw new PageException("meet IO Exception in buffer append,but we cannot understand it:" + e.getMessage());
+		}
+		buf.append(tempOutputStream);
+		pageList.add(tempOutputStream.transformToInputStream());
```

Comment:
adding page byte array into pageList should only perform when `cacheData` is true